src/brush.rs

```diff
@@ -147,6 +173,12 @@ impl Brush {
         self.draw(p);
     }
 
+    pub fn is_line_mode(&self) -> bool {
+        self.is_set(BrushMode::Line(LineDirection::Free))
+            || self.is_set(BrushMode::Line(LineDirection::Vertical))
+            || self.is_set(BrushMode::Line(LineDirection::Horizontal))
```

Comment:
Might be nice to use something like https://crates.io/crates/strum to make `LineDirection` runtime enumerable, but I don't know if we want to add a dependency just for this one spot.

Reply:
Yeah exactly, I tend to be very economical with my dependencies, as I like to keep build time low. You could however do:
```rust
self.modes.iter().any(|m| matches!(m, BrushMode::Line(_))
```

Reply:
`matches!` inspired me to rearrange some things. Should be a bit cleaner now.

Reply:
Indeed!